authentication/src/main/java/io/camunda/authentication/CamundaOidcUserService.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication;
+
+import io.camunda.authentication.entity.AuthenticationContext;
+import io.camunda.authentication.entity.CamundaOidcUser;
+import io.camunda.search.entities.MappingEntity;
+import io.camunda.service.AuthorizationServices;
+import io.camunda.service.MappingServices;
+import io.camunda.service.RoleServices;
+import io.camunda.service.TenantServices;
+import io.camunda.service.TenantServices.TenantDTO;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserRequest;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+
+@Service
+@Profile("auth-oidc")
+public class CamundaOidcUserService extends OidcUserService {
+  private static final Logger LOG = LoggerFactory.getLogger(CamundaOidcUserService.class);
+  private final MappingServices mappingServices;
+  private final TenantServices tenantServices;
+  private final RoleServices roleServices;
+  private final AuthorizationServices authorizationServices;
+
+  public CamundaOidcUserService(
+      final MappingServices mappingServices,
+      final TenantServices tenantServices,
+      final RoleServices roleServices,
+      final AuthorizationServices authorizationServices) {
+    this.mappingServices = mappingServices;
+    this.tenantServices = tenantServices;
+    this.roleServices = roleServices;
+    this.authorizationServices = authorizationServices;
+  }
+
+  @Override
+  public OidcUser loadUser(final OidcUserRequest userRequest) throws OAuth2AuthenticationException {
+    final OidcUser oidcUser = super.loadUser(userRequest);
+
+    final Map<String, Object> claims = userRequest.getIdToken().getClaims();
+    final List<MappingEntity> mappings = mappingServices.getMatchingMappings(claims);
+    final Set<Long> mappingKeys =
+        mappings.stream().map(MappingEntity::mappingKey).collect(Collectors.toSet());
+    if (mappingKeys.isEmpty()) {
+      LOG.debug("No mappings found for these claims: {}", claims);
+    }
+
+    return new CamundaOidcUser(
+        oidcUser,
+        mappingKeys,
+        new AuthenticationContext(
```

Comment:
🤔  I think if the mappingKeys come as empty, there is an pottential error here. I have tested with my OIDC user

```
11:34:53.136 [] [http-nio-0.0.0.0-8080-exec-5] [] ERROR org.apache.catalina.core.ContainerBase.[Tomcat].[localhost].[/].[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception
java.lang.NullPointerException: Expected a non-null query parameter for the hasChild query.
```

Would be the case to incldue a nullcheck over here?